### PR TITLE
Fix issue #6, enable gracefully shutdown for PMS and ADS

### PR DESCRIPTION
**Pull request by bill828**
Make two channels, one for Interrupt Signal, and the other one for error handling.

### CODE REVIEW

File: cmd/speedle-ads/main.go [hunk 1]
```diff
@@ -69,18 +73,54 @@ func main() {
 	evaluator, err := newEvaluator(conf)
 	if err != nil {
 		log.Panic(err)
-		return
+		os.Exit(1)
```
Comment (dingwenfang) [inline cmd/speedle-ads/main.go]: log.Panic() will call panic(), which will exit with a non-zero status.    
BTW, will we have different exit code for different error?

File: cmd/speedle-pms/main.go [hunk 2]
```diff
@@ -68,16 +72,49 @@ func main() {
...
+	log.Info("Stopping servers...")
+	// Stop all services
+	if httpServer != nil {
+		log.Info("Stopping HTTP Server.")
+		httpServer.Shutdown(context.Background())
+	}
+	if grpcServer != nil {
+		log.Info("Stopping GRPC Server.")
+		grpcServer.Stop()
+	}
```
Comment (dingwenfang) [inline cmd/speedle-pms/main.go]: exit with non-zero value?

File: cmd/speedle-ads/main.go (hunk 1 above)
Comment (bill828) [reply cmd/speedle-ads/main.go]: Done, use log.Fatal() instead.

File: cmd/speedle-pms/main.go (hunk 2 above)
Comment (bill828) [reply cmd/speedle-pms/main.go]: Done, if there was error occurred, exit with 1.

**Comment by dingwenfang** [comment]
The change looks OK to me now.

**Pull request merged**